ts/Core/Series/Series.ts

```diff
@@ -4374,6 +4378,21 @@ class Series {
             (series as any)[prop] = (preserve as any)[prop];
         });
 
+        // When switching back from sorted data to unsorted, reset indexes:
+        if (
+            options.data &&
+            options.dataSorting &&
+            options.dataSorting.enabled === false &&
+            oldOptions.dataSorting &&
+            oldOptions.dataSorting.enabled === true
```

Comment:
Do we need all the conditions? Maybe a less-than-perfect test like `if (options.data and 'dataSorting' in options)` would be good enough?

Reply:
It will be cost us some performance but will work, changed 👍